sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -136,19 +139,26 @@ export class MSFAANumberSharedService {
           updatedAt: now,
         },
       );
-      // Create the new MSFAA record for the student.
-      const newMSFAANumber = new MSFAANumber();
-      newMSFAANumber.msfaaNumber = await this.consumeNextSequence(
-        offeringIntensity,
-      );
-      newMSFAANumber.student = { id: studentId } as Student;
-      newMSFAANumber.referenceApplication = {
-        id: referenceApplicationId,
-      } as Application;
-      newMSFAANumber.offeringIntensity = offeringIntensity;
-      newMSFAANumber.creator = { id: auditUserId } as User;
-      newMSFAANumber.createdAt = now;
-      await entityManager.getRepository(MSFAANumber).save(newMSFAANumber);
+      let newMSFAANumber: MSFAANumber = null;
+      // There is not a MSFAA number to be re-activated.
+      if (!options?.msfaaNumber) {
+        // Create the new MSFAA record for the student.
+        newMSFAANumber = new MSFAANumber();
+        newMSFAANumber.msfaaNumber = await this.consumeNextSequence(
+          offeringIntensity,
+        );
+        newMSFAANumber.student = { id: studentId } as Student;
+        newMSFAANumber.referenceApplication = {
+          id: referenceApplicationId,
+        } as Application;
+        newMSFAANumber.offeringIntensity = offeringIntensity;
+        newMSFAANumber.creator = { id: auditUserId } as User;
+        newMSFAANumber.createdAt = now;
+        await entityManager.getRepository(MSFAANumber).save(newMSFAANumber);
+      } else {
```

Comment:
you can remove the else condition if you do , 
![image](https://github.com/bcgov/SIMS/assets/77353155/07e6206f-8aa2-46d5-aa8c-0c0ba5a809ab)